molviewspec/molviewspec/builder.py

```diff
@@ -673,55 +676,313 @@ def transparency(self, *, transparency: float = 0.8) -> Representation:
         return self
 
 
-class GenericVisuals(_Base):
+class Primitives(_Base):
     """
-    Experimental builder for custom, primitive visuals.
+    A collection of primitives (such as spheres, lines, ...) that will be grouped together and can be customized using
+    options.
     """
 
-    def sphere(
+    def mesh(
+        self,
+        *,
+        vertices: list[Vec3[float]],
+        indices: list[Vec3[int]],
+        colors: list[ColorT] | None = None,
+        # TODO should everything support `rotation`, just for convenience & consistency?
+    ) -> Primitives:
+        """
+        Construct custom meshes/shapes in a low-level fashion by providing vertices and indices.
+        :param vertices: collection of vertices
+        :param indices: collection of indices
+        :param colors: color value of each triangle
+        :return: this builder
+        """
+        params = make_params(MeshInlineParams, locals())
+        node = Node(kind="mesh", params=params)
+        self._add_child(node)
+        return self
+
+    # TODO mesh_from_uri, mesh_from_source
+
+    # TODO need ellipsis support
+    def circle(
         self,
         *,
-        position: tuple[float, float, float],
+        center: PositionT,
         radius: float,
-        color: ColorT,
-        label: str | None = None,
-        tooltip: str | None = None,
-    ) -> GenericVisuals:
+        segments: int | None = None,
+        theta_start: float = 0,
+        theta_length: float = math.pi * 2,
```

Comment:
What are the `theta_length` and `theta_start` exactly?

Reply:
Can be used to draw segments of a circle like a semicircle.